ractor/src/pg/mod.rs

```diff
@@ -138,6 +140,18 @@ pub fn join_scoped(scope: ScopeName, group: GroupName, actors: Vec<ActorCell>) {
             vacancy.insert(map);
         }
     }
+    match monitor.index.entry(scope.to_owned()) {
```

Comment:
so the problem that this approach takes (which is why it wasn't done this way in the beginning) is that this is technically a subtle race condition. We will probably need a global lock around both of the hash maps or something, but then there's going to be the benchmarking of the performance hit for taking a global lock.

Dashmaps are sharded maps so they can do concurrent updates from multiple threads, but what we'd be doing is locking around both maps with a singleton global lock.

Unless we can try and acquire both sharded locks, doing the updates, and then only releasing the locks simultaneously.

i.e.

```
lock(a)
lock(b)
update(a)
update(b)
release(a)
release(b)
```

It will be a little ugly with the scopes, but it's the best option for concurrent updates.

Reply:
You're right of course.
I hope to find the time to experiment and benchmark, can't currently promise a timeline, though :)

Reply:
Then again, as far as I understand we are only talking about the `map` and `index` fields which are only mutated in the `join*` and `leave*` functions.

It duplicates a bit more code but I think just locking both fields and then nesting their updates (so the locks get dropped simultaneously when leaving scope) might work.